cl/corpus_importer/management/commands/update_casenames_wl_dataset.py

```diff
@@ -0,0 +1,360 @@
+    :param docket_number: cleaned docket number from row
+    :param date_filed: formatted filed date from row
+
+    :return: list of matched OpinionCluster
+    """
+
+    citation_queries = Q()
+
+    for citation in valid_citations:
+        citation_query = Q(**citation) & Q(
+            cluster__docket__docket_number__contains=docket_number,
+            cluster__date_filed=date_filed,
+        )
+        citation_queries |= citation_query
+    possible_matches = Citation.objects.filter(
+        citation_queries
+    ).select_related("cluster")
+
+    return possible_matches
+
+
+def update_matched_case_name(
+    matched_cluster: OpinionCluster, csv_case_name: str
+) -> tuple[bool, bool]:
+    """Update case name of matched cluster and related docket if empty any of them
+
+    :param matched_cluster: OpinionCluster object
+    :param csv_case_name: case name from csv row
+    :return: tuple with boolean values if cluster and related docket case name updated
+    """
+    cluster_case_name_updated = False
+    docket_case_name_updated = False
+
+    if not matched_cluster.case_name:
+        # Save case name in cluster when we don't have it
+        matched_cluster.case_name = csv_case_name
+        matched_cluster.save()
+        logger.info(f"Case name updated for cluster id: {matched_cluster.id}")
+        cluster_case_name_updated = True
+
+    if not matched_cluster.docket.case_name:
+        # Save case name in docket when we don't have it
+        matched_cluster.docket.case_name = csv_case_name
+        matched_cluster.docket.save()
+        logger.info(
+            f"Case name updated for docket id: {matched_cluster.docket.id}"
+        )
+        docket_case_name_updated = True
+
+    return cluster_case_name_updated, docket_case_name_updated
+
+
+def process_csv(
+    filepath: str, delay: float, dry_run: bool, chunk_size: int
+) -> None:
+    """Process rows from csv file
+
+    :param filepath: path to csv file
+    :param delay: delay between saves in seconds
+    :param dry_run: flag to simulate update process
+    :param chunk_size: number of rows to read at a time
+    """
+
+    total_clusters_updated = 0
+    total_dockets_updated = 0
+
+    logger.info(f"Processing {filepath}")
+    for chunk in pd.read_csv(filepath, chunksize=chunk_size):
+        for row in chunk.dropna().itertuples():
+            (
+                index,
+                west_case_name,
+                court,
+                date_str,
+                cite1,
+                cite2,
+                docket,
+                volume,
+            ) = row
+
+            clean_docket_num = docket.strip('="').strip('"')
+            if not clean_docket_num:
+                logger.info(f"Row index: {index} - No docket number found.")
+                continue
+
+            date_filed = parse_date(date_str)
+            if not date_filed:
+                logger.info(f"Row index: {index} - No valid date found.")
```

Comment:
I think logging the errant date value would be valuable so it doenst have to be checked right?